yasa/hypno.py

```diff
@@ -474,3 +477,231 @@ def hypno_find_periods(hypno, sf_hypno, threshold="5min", equal_length=False):
+
+        ``trans_probas`` is a `right stochastic matrix
+        <https://en.wikipedia.org/wiki/Stochastic_matrix>`_, i.e. each row sums to 1.
+
+        If None (default), use transition probabilites from Metzner et al., 2021 [Metzner2021]_.
+        If :py:class:`pandas.DataFrame`, must have "from"-stages as indices and
+        "to"-stages as columns. Indices and columns must follow YASA integer
+        hypnogram convention (W = 0, N = 1, ...). Unscored/Artefact stages are not allowed.
+
+        .. note:: Transition probability matrices should indicate the transition
+            probability between *epochs* (i.e., probability of the next epoch) and
+            not simply stage (i.e., probability of non-similar stage).
+
+        .. seealso:: Return value from :py:func:`yasa.transition_matrix`
+    init_probas : :py:class:`pandas.Series` or None
+        Probabilites of each stage to initialize random walk.
+        If None (default), initialize with "From"-Wake row of ``trans_probas``.
+        If :py:class:`pandas.Series`, indices must be stages following YASA integer
+        hypnogram convention (see ``trans_probas``).
+    seed : int or None
+        Random seed for generating Markov sequence.
+
+    Returns
+    -------
+    hypno : np.ndarray
+        Hypnogram containing simulated sleep stages.
+
+    Notes
+    -----
+    Default transition probabilities can be found in the ``traMat_Epoch.npy`` file of
+    Supplementary Information for Metzner et al., 2021 [Metzner2021]_ (rounded values
+    are viewable in Figure 5b). Please cite this work if these probabilites are used
+    for publication.
+
+    References
+    ----------
+    .. [Metzner2021] Metzner, C., Schilling, A., Traxdorf, M., Schulze, H., & Krausse, P.
+                     (2021). Sleep as a random walk: a super-statistical analysis of EEG
+                     data across sleep stages. Communications Biology, 4.
+                     https://doi.org/10.1038/s42003-021-02912-6
+
+    Examples
+    --------
+    >>> from yasa import simulate_hypno
+    >>> hypno = simulate_hypno(tib=10, seed=0)
+    >>> hypno
+    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 2, 2, 4, 4, 4, 4, 4, 4, 4, 4]
+
+    Base the data off a real subject's transition matrix.
+
+    .. plot::
+        >>> import matplotlib.pyplot as plt
+        >>> import yasa
+        >>> url = (
+        >>>     "https://github.com/raphaelvallat/yasa/raw/master/"
+        >>>     "notebooks/data_full_6hrs_100Hz_hypno_30s.txt"
+        >>> )
+        >>> hypno = np.loadtxt(url)
+        >>> _, probas = yasa.transition_matrix(hypno)
+        >>> hypno_sim = yasa.simulate_hypno(tib=360, trans_probas=probas, seed=9)
+        >>> fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(7, 6))
+        >>> yasa.plot_hypnogram(hypno, ax=ax1)
+        >>> yasa.plot_hypnogram(hypno_sim, ax=ax2)
+        >>> ax1.set_title("True hypnogram")
+        >>> ax2.set_title("Simulated hypnogram")
+    """
+
```

Comment:
Empty line is not allowed after docstring. Can you please run `black yasa/hypno.py --line-length=100`?